pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -181,16 +181,10 @@ func mergeService(dst, src *model.Service, srcCluster string) {
 }
 
 // NetworkGateways merges the service-based cross-network gateways from each registry.
-func (c *Controller) NetworkGateways() map[string][]*model.Gateway {
-	gws := map[string][]*model.Gateway{}
+func (c *Controller) NetworkGateways() []*model.Gateway {
+	gws := make([]*model.Gateway, 0)
```

Comment:
very nit: it's more common to `var gws []*model.Gateway` and use `make` when we can set `len/cap`